### PR TITLE
Correct xctestrun ProductModuleName with dashes

### PR DESCRIPTION
**Pull request by mattrobmattrob**
In response to https://github.com/bazelbuild/rules_apple/pull/1907#pullrequestreview-1346288394, replicate https://github.com/google/xctestrunner/pull/42 in `ios_xctestrun_runner`.

### CODE REVIEW

**Comment by mattrobmattrob** [comment]
> you want to throw a dash in 1 of the tests to validate?

Ha. Was doing it locally as this was going through CI, yes! Up now.

File: test/ios_xctestrun_runner_ui_test.sh [hunk 1]
```diff
@@ -380,6 +389,18 @@ function test_ios_ui_test_underscore_with_filter() {
   expect_log "Executed 1 test, with 0 failures"
 }
 
+function test_ios_ui_test_dash_with_filter() {
+  create_sim_runners
+  create_ios_app
+  create_ios_ui_tests
+  do_ios_test --test_filter=PassingUITest/testPass2 //ios:PassingUITest-WithDash || fail "should pass"
```
[the remark this answers is not in the document]
Comment (mattrobmattrob) [reply test/ios_xctestrun_runner_ui_test.sh:396]: For `//ios:PassingUITest-WithDash`:
```
  <key>BazelTests</key>
  <dict>
    <key>ProductModuleName</key>
    <string>PassingUITest_WithDash</string>
```
For `//ios:PassingUITest_WithUnderscore`:
```
  <key>BazelTests</key>
  <dict>
    <key>ProductModuleName</key>
    <string>PassingUITest_WithUnderscore</string>
```

File: apple/testing/default_runner/ios_xctestrun_runner.template.sh [hunk 2]
```diff
@@ -271,14 +271,16 @@ if [[ "$should_use_xcodebuild" == true ]]; then
...
+
   readonly xctestrun_file="$test_tmp_dir/tests.xctestrun"
   /usr/bin/sed \
     -e "s@BAZEL_INSERT_LIBRARIES@$xctestrun_libraries@g" \
     -e "s@BAZEL_TEST_BUNDLE_PATH@__TESTROOT__/$test_bundle_name.xctest@g" \
     -e "s@BAZEL_TEST_ENVIRONMENT@$xctestrun_env@g" \
     -e "s@BAZEL_TEST_HOST_BASED@$xctestrun_test_host_based@g" \
     -e "s@BAZEL_TEST_HOST_PATH@$xctestrun_test_host_path@g" \
-    -e "s@BAZEL_TEST_PRODUCT_MODULE_NAME@$test_bundle_name@g" \
+    -e "s@BAZEL_TEST_PRODUCT_MODULE_NAME@$test_product_module_name@g" \
```
Comment (keith) [inline apple/testing/default_runner/ios_xctestrun_runner.template.sh:283]:
```suggestion
    -e "s@BAZEL_TEST_PRODUCT_MODULE_NAME@${test_bundle_name//-/_}@g" \
```

the magic of bash

Comment (mattrobmattrob) [reply apple/testing/default_runner/ios_xctestrun_runner.template.sh:283]: You'd think I'd remember this after using bash substitution multiple times in the past week.

**Review by keith — APPROVED** [review]

**Pull request merged**